apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -974,9 +985,12 @@ stream(Data, #state{pid=Pid, size=undefined, buf=Buf}=State) ->
     case binary:split(Buf0, <<"\n">>) of
         [SizeBin, Tail] ->
             Size = binary_to_integer(SizeBin),
+            prometheus_counter:inc(mesos_listener, messages_total, [], 1),
+            prometheus_counter:inc(mesos_listener, bytes_total, [], Size),
             State0 = State#state{size=Size, buf= <<>>},
             stream(Tail, State0);
         [Buf0] when byte_size(Buf0) > 12 ->
+            prometheus_counter:inc(mesos_listener, failures_total, [], 1),
```

Comment:
This one is trickier. Let me think how to implement it better, I guess we should increment it in `terminate/2` and when we receive an unexpected http response from mesos.

Reply:
Do you have any new suggestion here?

Reply:
I'm sorry, I haven't had a chance to check it. I still think `terminate/2` and http errors are the best places for this metric. Could you please try to add a fake `handle_call` method that always stops the process with a non-`normal` reason? The metric must be incremented in this case.